optuna/samplers/_tpe/sampler.py

```diff
@@ -155,13 +181,67 @@ def infer_relative_search_space(
         self, study: Study, trial: FrozenTrial
     ) -> Dict[str, BaseDistribution]:
 
-        return {}
+        if not self._multivariate:
+            return {}
+
+        search_space = {}  # type: Dict[str, BaseDistribution]
+        for name, distribution in self._search_space.calculate(study).items():
+            if not isinstance(
+                distribution,
+                (
+                    distributions.UniformDistribution,
+                    distributions.LogUniformDistribution,
+                    distributions.DiscreteUniformDistribution,
+                    distributions.IntUniformDistribution,
+                    distributions.IntLogUniformDistribution,
+                    distributions.CategoricalDistribution,
+                ),
+            ):
+                continue
+            search_space[name] = distribution
+
+        return search_space
 
     def sample_relative(
         self, study: Study, trial: FrozenTrial, search_space: Dict[str, BaseDistribution]
     ) -> Dict[str, Any]:
 
-        return {}
+        if search_space == {}:
+            return {}
+
+        param_names = list(search_space.keys())
+        values, scores = _get_multivariate_observation_pairs(study, param_names)
+
+        # If the number of samples is insufficient, we run random trial.
+        n = len(scores)
+        if n < self._n_startup_trials:
+            ret = {}
+            for param_name, param_distribution in search_space.items():
+                ret[param_name] = self._random_sampler.sample_independent(
+                    study, trial, param_name, param_distribution
+                )
+            return ret
```

Comment:
If we return `{}`, parameters are automatically sampled by `sample_independent`. To simplify the code, how about doing it?
```suggestion
        if n < self._n_startup_trials:
            return {}
```

Reply:
Fixed as you suggested.